crates/stages/src/stages/sender_recovery.rs

```diff
@@ -83,43 +82,49 @@ impl<DB: Database> Stage<DB> for SenderRecoveryStage {
         // Acquire the cursor for inserting elements
         let mut senders_cursor = tx.cursor_write::<tables::TxSenders>()?;
 
-        // Query the transactions from both database and static files
-        let transactions = provider.raw_transactions_by_tx_range(tx_range.clone())?;
-
         // Iterate over transactions in chunks
         info!(target: "sync::stages::sender_recovery", ?tx_range, "Recovering senders");
 
-        // channels used to return result of sender recovery.
-        let mut channels = Vec::new();
-
         // Spawn recovery jobs onto the default rayon threadpool and send the result through the
         // channel.
         //
-        // We try to evenly divide the transactions to recover across all threads in the threadpool.
-        // Chunks are submitted instead of individual transactions to reduce the overhead of work
-        // stealing in the threadpool workers.
-        let chunk_size = self.commit_threshold as usize / rayon::current_num_threads();
-        // prevents an edge case
-        // where the chunk size is either 0 or too small
-        // to gain anything from using more than 1 thread
-        let chunk_size = chunk_size.max(16);
-
-        for chunk in &tx_range.zip(transactions).chunks(chunk_size) {
+        // Transactions are different size, so chunks will not all take the same processing time. If
+        // chunks are too big, there will be idle threads waiting for work. Choosing an
+        // arbitrary smaller value to make sure it doesn't happen.
+        let chunk_size = 100;
```

Comment:
hmm, shouldn't we somehow depend on the number of available threads anyway?

Reply:
tbh it's hard to get the heuristic right i think. 

less is more in this case, thats my original point, but open to suggestions